source/extensions/filters/udp/dns_filter/dns_parser.cc

```diff
@@ -193,8 +193,11 @@ bool DnsMessageParser::parseDnsObject(DnsQueryContextPtr& context,
     }
   } while (!done);
 
-  if (!context->header_.flags.qr && context->header_.answers) {
-    ENVOY_LOG(debug, "Answer records present in query");
+  // Only QR == 0 and questions without any answer RRs are expected
+  if (!(context->header_.flags.qr == 0 && context->header_.answers == 0 &&
```

Comment:
It looks to me there is no test in dns_filter_test.cc to test this new logic. Could you please add new test cases under DnsFiltertest to verify if in case such cases are detected (any tpe of answers records exists in query message,), and the code behave as expected.

Reply:
Done.